nautobot/core/tests/test_views.py

```diff
@@ -452,6 +459,155 @@ def test_get_object_with_constrained_permission(self):
+            "LOG_DEPRECATION_WARNINGS",
+            "LOG_LEVEL",
+            "MEDIA_URL",
+            "MESSAGE_TAGS",
+            "MIDDLEWARE",
+            "PROMETHEUS_EXPORT_MIGRATIONS",
+            "REMOTE_AUTH_AUTO_CREATE_USER",
+            "REMOTE_AUTH_HEADER",
+            "REST_FRAMEWORK",
+            "REST_FRAMEWORK_ALLOWED_VERSIONS",
+            "REST_FRAMEWORK_VERSION",
+            "ROOT_URLCONF",
+            "SECURE_PROXY_SSL_HEADER",
+            "SESSION_CACHE_ALIAS",
+            "SESSION_ENGINE",
+            "SHELL_PLUS_DONT_LOAD",
+            "SHORT_TIME_FORMAT",
+            "SILKY_ANALYZE_QUERIES",
+            "SILKY_AUTHENTICATION",
+            "SILKY_AUTHORISATION",
+            "SILKY_INTERCEPT_FUNC",
+            "SILKY_PERMISSIONS",
+            "SILKY_PYTHON_PROFILER",
+            "SILKY_PYTHON_PROFILER_BINARY",
+            "SILKY_PYTHON_PROFILER_EXTENDED_FILE_NAME",
+            "SOCIAL_AUTH_BACKEND_PREFIX",
+            "SOCIAL_AUTH_POSTGRES_JSONFIELD",
+            "SPECTACULAR_SETTINGS",
+            "STATICFILES_DIRS",
+            "STATIC_URL",
+            "TEMPLATES",
+            "TESTING",
+            "TEST_RUNNER",
+            "USE_I18N",
+            "USE_TZ",
+            "USE_X_FORWARDED_HOST",
+            "VERSION",
+            "VERSION",
+            "VERSION_MAJOR",
+            "VERSION_MINOR",
+            "WEBSERVER_WARMUP",
+            "WSGI_APPLICATION",
+            "X_FRAME_OPTIONS",
+        ]
+
+        # Retrieve all variables from settings.py.
+        # and trim out all noises by retaining variables with only uppercase letters and "_".
+        existing_settings_variables = list(settings.__dict__.keys())
+        existing_settings_variables = [
+            word for word in existing_settings_variables if all([letter in ascii_uppercase + "_" for letter in word])
+        ]
+
+        # Some of the variables in the JSON schema are contained in the CONSTANCE_CONFIG setting variable, e.g. BANNER_TOP, BANNER_BOTTOM and etc.
+        existing_constance_config_variables = list(settings.__dict__.get("CONSTANCE_CONFIG").keys())
+        # All the documented settings variable in the JSON schema from settings.json
+        existing_json_schema_variables = list(self.json_data["properties"].keys())
+
+        # Check if there is any unsupported setting variable in settings.py.
+        for variable in existing_settings_variables:
+            if variable not in existing_json_schema_variables + ALLOWED_SETTINGS:
+                self.fail(f"Unsupported settings variable {variable} detected in nautobot/core/settings.py")
+
+        # Check if there is any unsupported setting variable in settings.json.
+        for variable in self.json_data["properties"].keys():
+            if variable not in existing_settings_variables + existing_constance_config_variables + ALLOWED_SETTINGS:
+                self.fail(f"Unsupported settings variable {variable} detected in nautobot/core/settings.json")
```

Comment:
```suggestion
        # Check if there is any undocumented setting variable in settings.py.
        expected_variables = set(existing_json_schema_variables) & set(ALLOWED_SETTINGS)
        for variable in existing_settings_variables:
            if variable not in expected_variables:
                self.fail(f"Undocumented settings variable {variable} detected in nautobot/core/settings.py")

        # Check if there is any nonexistent settings variable in settings.json.
        expected_variables = set(existing_settings_variables) & set(existing_constance_config_variables) & set(ALLOWED_SETTINGS)
        for variable in self.json_data["properties"].keys():
            if variable not in expected_variables:
                self.fail(f"Nonexistent settings variable {variable} detected in nautobot/core/settings.json")
```

Reply:
The combinator here `set(existing_json_schema_variables) & set(ALLOWED_SETTINGS)` should be "||" right?

Reply:
Yeah, sorry :-)  Good catch.